### ISSUE TITLE
NWaves and Torchaudio are producing different outputs

### ISSUE THREAD

**Issue by xhy2008** [ar1st0crat/NWaves#91]
Hello,I wanted to run a music beat detection model with ONNX in my Godot project,which looks like muse-dash. The model was designed to accept mel-spectrogram and produce a list of the notes.
I wrote both torchaudio and NWaves code to generate mel-spectrogram,and caculate the checksum.
I found that the checksum was really different,so I searched the wiki,but nothing seemed to work.
If you finds my code wrong,just tell me.
Here's the code.
Python:
`
import torch
import torchaudio
import os
from torchaudio.transforms import Resample

def convert_audio_to_mel(audio_path, target_sr=16000):
    waveform, orig_sr = torchaudio.load(audio_path)
    
    # convert to mono
    if waveform.shape[0] > 1: 
        waveform = torch.mean(waveform, dim=0, keepdim=True)
    
    # resample
    if orig_sr != target_sr:
        resampler = Resample(orig_sr, target_sr)
        waveform = resampler(waveform)
    # mel    
    melspec_transform = torchaudio.transforms.MelSpectrogram(
        sample_rate=target_sr,
        n_fft=2048,
        win_length=None,
        hop_length=512,
        power=2.0,
        n_mels=64,
        center=False,
        norm=None, 
        mel_scale='htk',
        window_fn=torch.hann_window
    )
    
    melspec = melspec_transform(waveform)
    log_S = 10 * torch.log10(torch.clamp(melspec, min=1e-10))
    base_name = os.path.basename(audio_path)
    print(base_name, "   processed")
    return log_S.numpy()

total=0.0
melspec = convert_audio_to_mel('./test_mel.mp3')
print(melspec.shape)#[1,64,156480]
for i in melspec[0]:
      for j in i:
            total+=j
print("checksum:",total)
`

C#:
`
using NWaves.Audio;
using NWaves.FeatureExtractors;
using NWaves.FeatureExtractors.Options;
using NWaves.Filters.Fda;
using NWaves.Signals;
using NWaves.Windows;
using NAudio.Wave;
using NAudio.Dsp;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using System.Linq;
using System.IO;
using System;
public class MelProcessor
{
    public static DiscreteSignal LoadAndConvertMp3(string mp3Path)
    {
        using (var mp3Reader = new Mp3FileReader(mp3Path))
        {
            var pcmStream = WaveFormatConversionStream.CreatePcmStream(mp3Reader);

            var targetFormat = new NAudio.Wave.WaveFormat(16000, 1);

            // resample
            using (var resampler = new MediaFoundationResampler(pcmStream, targetFormat))
            {
                resampler.ResamplerQuality = 60; 
                var sampleProvider = resampler.ToSampleProvider();
                var sampleList = new List<float>();
                float[] buffer = new float[4096]; 
                int read;
                while ((read = sampleProvider.Read(buffer, 0, buffer.Length)) > 0)
                {
                    sampleList.AddRange(buffer.Take(read));
                }
                var samples = sampleList.ToArray(); 
                return new DiscreteSignal(16000, samples);
            }
        }
    }
    public void ProcessMp3()
    {
        var signal =LoadAndConvertMp3("test_mel.mp3");
        var fftSize = 2048;
        var hopSize = 512;
        var melCount = 64;
        var mfccExtractor = new FilterbankExtractor(
        new FilterbankOptions
        {
            SamplingRate = 16000,
            FrameSize = fftSize,
            FftSize = fftSize,
            HopSize = hopSize,
            Window = WindowType.Hann,
            FilterBank = FilterBanks.Triangular(fftSize, 16000, 
                FilterBanks.MelBands(melCount, 16000)),
            SpectrumType=SpectrumType.Power
        });
        var spectrogram = mfccExtractor.ComputeFrom(signal);
        var logSpectrogram = spectrogram
            .Select(frame => frame
                .Select(v => 10 * Math.Log10(Math.Max(v, 1e-10)))
                .ToArray())
            .ToList();
        Console.WriteLine(logSpectrogram[0][0]);
    }
`
